subprojects/dependency-management/src/main/java/org/gradle/internal/resource/transfer/DefaultCacheAwareExternalResourceAccessor.java

```diff
@@ -76,8 +78,8 @@ public LocallyAvailableExternalResource getResource(final URI location, final Re
             return copyToCache(location, fileStore, delegate.withProgressLogging().getResource(location, false));
         }
 
-        // We might be able to use a cached/locally available version
-        if (cached != null && !externalResourceCachePolicy.mustRefreshExternalResource(getAgeMillis(timeProvider, cached))) {
+        // We might be able to use a cached/locally available version, when offline return the cached value
```

Comment:
I'm not sure about the `|| isOffline` bit as part of this change set, as it is a semantically breaking change IIUC.